deploy.dog/InstallOrUpdate.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Determine if our key file exists as this determine if we have a key already (and so run an upgrade) or need to do a new install
+if [ ! -f ../../../shared/deploy.dog.cachet-key ]; then
+    # File not found, new install
+    echo "New Install"
+
+    # Generate new key
+    KEY=$(php artisan key:generate --show)
+    KEY=(${KEY[@]})
+
+    # Add key into .env file
+    sed -i "s/DDREPLACECACHETKEY/$(echo $KEY | sed -e 's/\\/\\\\/g; s/\//\\\//g; s/&/\\\&/g')/" .env
+
+    # Install Cachet
+    php artisan app:install
+
+    # Store key for future
+    echo "$KEY" > ../../../shared/deploy.dog.cachet-key
+
+else
+    # File found, we're upgrading
+    echo "Upgrading"
+
+    # Add key into .env file
+    KEY=$(head -n 1 ../../../shared/deploy.dog.cachet-key)
+    sed -i "s/DDREPLACECACHETKEY/$(echo $KEY | sed -e 's/\\/\\\\/g; s/\//\\\//g; s/&/\\\&/g')/" .env
+
+    # Upgrade Cachet
+    php artisan down
+    php artisan app:update
```

Comment:
Don't forget `php artisan up`.

Reply:
Thanks James. I've put the `php artisan up` bit in the `postSymlinkChangeTasks` section so that will mean that the site is in maintenance mode for very slightly longer, but we can ensure that when it comes back up, the code running will be the newly deployed version. See lines 10-12 and 29 of the `deploy.dog.yml` file. It might be better off in the `InstallOrUpdate.sh` file though as it does mean that that file is not the complete process in itself (which people might expect it to be)?

In fact, I don't think the `php artisan down` bit is quite right, sorry, as this will be running from the about-to-be-released directory and it should be within the currently running version (otherwise it's not going to have any impact). I'd overlooked how the up/down commands [actually work](https://www.reddit.com/r/laravel/comments/311i4q/what_is_php_artisan_down_actually_doing/).

I'll push a fix for this...

Reply:
Ok, that last push should solve this. Sorry about that.